### PR TITLE
Rework cudf::strings::from_booleans to use make_strings_children

### PR DESCRIPTION
**Pull request by vuule**
## Description

Issue https://github.com/rapidsai/cudf/issues/12167
 
Use `make_strings_children` in `cudf::strings::from_booleans` to simplify code and make the function more robust.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by vuule** [comment]
CC @davidwendt

**Review by davidwendt — APPROVED** [review]

File: cpp/src/strings/convert/convert_booleans.cu [hunk 1]
```diff
@@ -90,6 +90,32 @@ std::unique_ptr<column> to_booleans(strings_column_view const& strings,
...
+
+  __device__ void operator()(size_type idx) const
+  {
+    if (d_column.is_null(idx)) {
+      if (d_chars == nullptr) d_offsets[idx] = 0;
+      return;
+    }
+
+    if (d_chars != nullptr) {
+      auto const result = d_column.element<bool>(idx) ? d_true : d_false;
```
Comment (divyegala) [inline cpp/src/strings/convert/convert_booleans.cu:110]: This doesn't need template keyword after the dot operator to compile?

Comment (davidwendt) [reply cpp/src/strings/convert/convert_booleans.cu:110]: No. You generally only need that if the function this is called in is also templated in order to help the compiler resolve the typenames correctly.

**Review by divyegala — APPROVED** [review]

**Comment by vuule** [comment]
@gpucibot merge

**Pull request merged**